okio-files/src/commonMain/kotlin/okio/Filesystem.kt

```diff
@@ -22,4 +22,12 @@ abstract class Filesystem {
    * are resolved against when using this filesystem.
    */
   abstract val cwd: Path
+
+  companion object {
+    /**
+     * The current process's host filesystem. Use this instance directly, or dependency inject a
+     * [Filesystem] to make code testable.
+     */
+    val SYSTEM: Filesystem = PLATFORM_FILESYSTEM
```

Comment:
Not a huge fan of `Filesystem.SYSTEM` at a potential call site (feels repetitive), but can't think of a really good alternative. Referencing the platform specific file system through just `Filesystem` would be nice, but not sure how to achieve it. I wanna look into the coroutines lib for inspiration on modeling platform-specific singletons (e.g. `Dispatchers.Main`) and come back with better suggestions!

Reply:
You're supposed to call through the abstraction. Referencing an implementation at a callsite is supposed to look bad because it's an anti-pattern.

Reply:
Makes sense

Reply:
I'm pattern matching `Dns.SYSTEM` from OkHttp which seemed like a good name for something provided by the OS.

Reply:
Also TimeSouce.SYSTEM from kotlinx.datetime!

Reply:
I'm sold!